shanoir-ng-front/src/app/studies/study-history/study-history.component.ts

```diff
@@ -0,0 +1,87 @@
+import {StudyUser} from "../shared/study-user.model";
+import {Observable} from "rxjs";
+
+@Component({
+  selector: 'study-history',
+  templateUrl: './study-history.component.html',
+  styleUrl: './study-history.component.css'
+})
+export class StudyHistoryComponent {
+
+    @ViewChild('table', {static: false}) table: TableComponent;
+    @Input() study: Study;
+    @Input() eventHistory: Promise<any>;
+
+    history: ShanoirEvent[]=[];
+
+    historyColumns: ColumnDefinition[] = [
+        {headerName: 'Creation date', field: 'creationDate', type: 'dateTime'},
+        {headerName: 'User', field: 'username'},
+        {headerName: 'Event type', field: 'eventType', cellRenderer: function (params: any) {
+                if (params.data.eventType.includes(".event")) {
+                    params.data.eventType = params.data.eventType.replace(".event", "");
+                }
+                if (params.data.eventType.includes("userAddToStudy")) {
+                    params.data.message = "User " + params.data.username + " access granted"
+                }
+                return params.data.eventType;
+            }
+        },
+        {headerName: 'ObjectId', field: 'objectId', route: function(params:ShanoirEvent) {
+            let event = params.eventType;
+            let id = params.objectId;
+            if (event.includes("create")) {
+                if (event.includes("Dataset") && !event.includes("Acquisition")) { return "/dataset/details/" + id; }
+                else if (event.includes("Examination")) { return "/examination/details/" + id; }
+                else if (event.includes("Acquisition")) { return "/dataset-acquisition/details/" + id; }
+                else if (event.includes("Subject")) { return "/subject/details/" + id; }
+            } else if (event.includes("update")) {
+                if (event.includes("Study")) { return "/study/details/" + id; }
+                else if (event.includes("Examination")) { return "/examination/details/" + id; }
+                else if (event.includes("Subject")) { return "/subject/details/" + id; }
+            } else if (event.includes("import")) {
+                if (event.includes("Dataset")) params.objectId = null;
+            } else {
+                if (event.includes("userAddToStudy")) {
+                    params.objectId = null;
+                }
+            }
+        }},
+        {headerName: 'Message', field: 'message'}
+    ];
+
+    constructor(
+        private shanoirEventService: ShanoirEventService
+    ) {}
+    ngOnInit() {
+        this.eventHistory.then( () => this.onFetchHistory());
+    }
+
+    public onFetchHistory() {
+        this.shanoirEventService.requestHistory(this.study.id).then(history => {
+
+            history.forEach(item => {
+                let studyUser : StudyUser;
+                studyUser = this.study.studyUserList.find(user => user.userId == item.userId);
```

Comment:
Here you get the user for every event, maybe you could create a Map<userId,username> at the beginning of the method then  re-use it when iterating over the events.